diff --git a/lib/curl_share.c b/lib/curl_share.c
index 386a2b5..3576250 100644
--- a/lib/curl_share.c
+++ b/lib/curl_share.c
@@ -368,7 +368,7 @@ CURLSHcode curl_share_cleanup(CURLSH *sh)
   return CURLSHE_OK;
 }
 
-CURLSHcode Curl_share_lock(struct Curl_easy *data, curl_lock_data type,
+CURLSHcode __stdcall Curl_share_lock(struct Curl_easy *data, curl_lock_data type,
                            curl_lock_access accesstype)
 {
   struct Curl_share *share = data->share;
@@ -385,7 +385,7 @@ CURLSHcode Curl_share_lock(struct Curl_easy *data, curl_lock_data type,
   return CURLSHE_OK;
 }
 
-CURLSHcode Curl_share_unlock(struct Curl_easy *data, curl_lock_data type)
+CURLSHcode __stdcall Curl_share_unlock(struct Curl_easy *data, curl_lock_data type)
 {
   struct Curl_share *share = data->share;
 
diff --git a/lib/curl_share.h b/lib/curl_share.h
index 69001be..9b88047 100644
--- a/lib/curl_share.h
+++ b/lib/curl_share.h
@@ -76,9 +76,9 @@ struct Curl_share {
 #endif
 };
 
-CURLSHcode Curl_share_lock(struct Curl_easy *data, curl_lock_data type,
+CURLSHcode __stdcall Curl_share_lock(struct Curl_easy *data, curl_lock_data type,
                            curl_lock_access accesstype);
-CURLSHcode Curl_share_unlock(struct Curl_easy *data, curl_lock_data type);
+CURLSHcode __stdcall Curl_share_unlock(struct Curl_easy *data, curl_lock_data type);
 
 /* convenience macro to check if this handle is using a shared SSL spool */
 #define CURL_SHARE_ssl_scache(data) ((data)->share &&                    \
